File: GXpert/GXpert.Web/Modules/ServerTypes/Skills/SkillCategoryForm.ts
import { StringEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SkillCategoryForm {
    Name: StringEditor;
    Description: TextAreaEditor;
}

export class SkillCategoryForm extends PrefixedContext {
    static readonly formKey = 'Skills.SkillCategory';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SkillCategoryForm.init)  {
            SkillCategoryForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;

            initFormType(SkillCategoryForm, [
                'Name', w0,
                'Description', w1
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Skills/SkillCategory/SkillCategoryForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Skills.Forms;

[FormScript("Skills.SkillCategory")]
[BasedOnRow(typeof(SkillCategoryRow), CheckNames = true)]
public class SkillCategoryForm
{
    public string Name { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
  
}